src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,163 @@
+          ratings.add(Double.parseDouble(split[1]));
+          this.totalReviews++;
+        }
+      }
+      this.productsMap = buildAndSetMap(products);
+      this.usersMap = buildAndSetMap(users);
+      this.totalProducts = productsMap.size();
+      this.totalUsers = usersMap.size();
+
+      writeToDataModelFile("./dataModel.csv", products, users, ratings);
+    } catch(IOException e) {
+      System.out.println("Exception reading line.");
+    }
+  }
+
+  public static BufferedReader readGzippedFile(String filePath) throws IOException, FileNotFoundException {
+    InputStream fileStream = new FileInputStream(filePath);
+    InputStream gzipStream = new GZIPInputStream(fileStream);
+    Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+    BufferedReader buffered = new BufferedReader(decoder);
+    return buffered;
+  }
+
+  public Map<String,Integer> buildAndSetMap(ArrayList<String> list) {
+    Map<String,Integer> map = new HashMap<String,Integer>();
+    int id = 1;
+    for(String s:list) {
+      if(!map.containsKey(s)) {
+        map.put(s, id);
+        id++;
+      }
+    }
+    return map;
+  }
+
+  public void writeToDataModelFile(String filePath, ArrayList<String>prod, ArrayList<String> usr, ArrayList<Double> rat) {
+    try {
+      PrintWriter pw = new PrintWriter(new File(filePath));
+
+      for(int i = 0; i < prod.size(); i++) {
+        // String lineToPrint = String.format("%s,%s,%s", usr.get(i), productsMap.get(prod.get(i)), rat.get(i));
+        // pw.write(lineToPrint);
+        pw.write(usersMap.get(usr.get(i))+","+productsMap.get(prod.get(i))+","+rat.get(i));
+        pw.println();
+      }
+
+      pw.close();
+    } catch(IOException e) {
+
+    }
+  }
+
+  public int getTotalReviews() {
+    return this.totalReviews;
+  }
+
+  public int getTotalProducts() {
+    return this.totalProducts;
+  }
+
+  public int getTotalUsers() {
+    return this.totalUsers;
+  }
+
+  public List<String> getRecommendationsForUser(String userId) {
+    List<String> recs = new ArrayList<String>();
+    DataModel model = null;
+  	UserSimilarity similarity = null;
+  	List<RecommendedItem> recommendations = null;
+
+    try {
+      model = new FileDataModel(new File("dataModel.csv"));
+    } catch(IOException e) {
+
+    }
+
+    try {
+      similarity = new PearsonCorrelationSimilarity(model);
+    } catch(TasteException e) {
+
+    }
+
+    UserNeighborhood neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+    UserBasedRecommender recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+
+    try {
+      recommendations = recommender.recommend(usersMap.get(userId), 3);
+    } catch(TasteException e) {
+
+    }
+
+    for(RecommendedItem recommendation : recommendations) {
+      String rec = getKeyFromProdMap((int) recommendation.getItemID());
+      recs.add(rec);
+    }
+
+    return recs;
+
+  }
+
+  public String getKeyFromProdMap(int value) {
```

Comment:
In this function, what happens if there is two keys with the same value?

For example, if I have the keys->value: A->2, B->4 and C->2 and I want to know the key for the value 2, which key it returns to me? Remember you're using a HashMap implementation :)